### PR TITLE
Fixing non targted already started resources

### PR DESCRIPTION
**Pull request by ankitpws**
# Description
These recommendations are intended to promote code quality and team communication during software development. They cover a variety of topics, including ensuring that pull requests are submitted to the correct branch, documenting new methods, preserving consistency across many services, and avoiding typical blunders like accessing APIs or DB queries within loops. Sensitive data should not be uploaded, and changes to environment variables or database models should be executed consistently. Teams may work more effectively and develop higher-quality software by adhering to these standards.

## Type of change
Please choose appropriate options.

- [ ]  Bug fix (non-breaking change which fixes an issue)
- [ ]  New feature (non-breaking change which adds functionality)
- [ ]  Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ]  Enhancement (additive changes to improve performance)
- [ ]  This change requires a documentation update

## Checklist 

- [ ]  It's critical to avoid making needless file modifications in contributions, such as adding new lines, console logs, or additional spaces, to guarantee cleaner and more efficient code. Furthermore, eliminating unnecessary imports from a file might enhance code readability and efficiency.
- [ ]  Ensure that the pull request is assigned to the right base branch and that the development branch name contains the JIRA Task Id. Furthermore, each commit message should include the JIRA Task Id in the manner "ED-100: message".
- [ ]  Only update packages if it is mentioned and authorized in the design document, and make sure that you have the required permissions.
- [ ]  Avoid making API and database queries inside a loop as it can lead to performance issues and slow down the system.
- [ ]  When calling another function inside a given function, add comments explaining the purpose and meaning of the passed arguments and expected return values.
- [ ]  If adding a blank argument in a function, add a comment explaining the reason for the blank argument.
- [ ]  Before submitting a pull request, do a self-review of your code to ensure there are no conflicts with the base branch and all comments have been addressed.
- [ ]  Before merging a pull request, it's important to have other team members review it to catch any potential errors or issues
- [ ]  To maintain code integrity, it's important to remove all related changes when removing code during a code review.
- [ ]  If new constants, endpoints, or utility functions are introduced, it is important to check if they already exist in the service to avoid any duplication.
- [ ]  Whenever a new environment variable is added to a service, it's important to ensure that the necessary changes are made to related files such as ".env.sample" and "envVariables.js" to maintain consistency and avoid errors. Additionally, the new environment variable should be added to the devops repository to ensure that it is properly documented and accessible to the team.
- [ ]  When adding a new function to a service, it is important to document it with relevant information such as the name, parameters, and return value in a consistent format across all services. Additionally, if there are any changes to the API response, ensure that the documentation in the controllers is updated accordingly. 
- [ ]  Write a clear and concise commit message that describes the changes made.
- [ ]  Maintain consistent function signature and code across all services when adding a function to multiple services. Implement changes to database models in all services that use the same model. 
- [ ]  Use only let and const. Do not use var.
- [ ]  Make common functions for repetitive code blocks.
- [ ]  Avoid uploading sensitive information such as secret tokens or passwords in pull requests to ensure data security.
- [ ]  Maintain consistent indentation and spacing throughout the code.

### CODE REVIEW

File: controllers/v1/surveysController.js [hunk 1]
```diff
@@ -871,4 +869,101 @@ module.exports = class Surveys extends Abstract {
...
+    /**
+   * Get survey and observation documents for program.
+   * @method
+   * @name getImportedSurveysAndObservations
+   * @param {Object} req -request Data.
+   * @param {String} req.params._id - programId.
+   * @returns {JSON} 
+   */
+
+    async getImportedSurveysAndObservations(req,res){
```
Comment (aks30) [inline controllers/v1/surveysController.js:934]: @ankitpws Please create different APIs for observation and survey

File: module/observations/helper.js [hunk 2]
```diff
@@ -2255,6 +2255,48 @@ module.exports = class ObservationsHelper {
...
+      * @returns {result} - all the observation which user has started in that program. 
+     */
+
+    static getImportedObservations(userId, programId){
+        return new Promise(async (resolve, reject) => {
+            try {
+                
+                let surveyData = await this.observationDocuments({
+                    programId : programId,
+                    createdBy : userId,
```
Comment (aks30) [inline module/observations/helper.js:2274]: @ankitpws the condition sequence has to be reversed, createdBy and then programId, it makes the query faster

File: module/surveys/helper.js [hunk 3]
```diff
@@ -1888,4 +1888,44 @@ module.exports = class SurveysHelper {
...
+      * @param  {String} programId - program Id.
+      * @returns {result} - all the survey which user has started in that program. 
+     */
+    static getImportedSurveys(userId, programId){
+        return new Promise(async (resolve, reject) => {
+            try {
+                
+                let surveyData = await this.surveyDocuments({
+                    programId : programId,
+                    createdBy : userId,
```
Comment (aks30) [inline module/surveys/helper.js:1906]: @ankitpws  the condition sequence has to be reversed, createdBy, and then programId, it makes the query faster

**Review by aks30 — CHANGES_REQUESTED** [review]
Reviewed on 31 May, 3:30 PM

File: module/surveys/helper.js (hunk 3 above)
Comment (ankitpws) [reply module/surveys/helper.js:1903]: Done

File: module/observations/helper.js (hunk 2 above)
Comment (ankitpws) [reply module/observations/helper.js:2271]: done

File: controllers/v1/surveysController.js (hunk 1 above)
Comment (ankitpws) [reply controllers/v1/surveysController.js]: done

File: controllers/v1/observationsController.js [hunk 4]
```diff
@@ -2289,6 +2287,78 @@ module.exports = class Observations extends Abstract {
...
+    /**
+   * Get observation documents for program.
+   * @method
+   * @name getStartedObservations
+   * @param {Object} req -request Data.
+   * @param {String} req.params._id - programId.
+   * @returns {JSON} 
+   */
+
+    async getStartedObservations(req,res){
```
Comment (aks30) [inline controllers/v1/observationsController.js:2333]: @ankitpws Let's move this new API under the "users" controller with the name "observations"

File: controllers/v1/surveysController.js [hunk 5]
```diff
@@ -871,4 +868,77 @@ module.exports = class Surveys extends Abstract {
...
+    /**
+   * Get survey documents for program.
+   * @method
+   * @name getStartedSurveys
+   * @param {Object} req -request Data.
+   * @param {String} req.params._id - programId.
+   * @returns {JSON} 
+   */
+
+    async getStartedSurveys(req,res){
```
Comment (aks30) [inline controllers/v1/surveysController.js:914]: @ankitpws Let's move this new API under the "users" controller with the name "surveys"

File: module/observations/helper.js [hunk 6]
```diff
@@ -2255,6 +2253,48 @@ module.exports = class ObservationsHelper {
...
+ /**
+      * Get imported observation documents.
+      * @method
+      * @name getStartedObservations
+      * @param  {String} userId - userId of user.
+      * @param  {String} programId - program Id.
+      * @returns {result} - all the observation which user has started in that program. 
+     */
+
+    static getStartedObservations(userId, programId){
```
Comment (aks30) [inline module/observations/helper.js:2266]: @ankitpws this will be moved to usersHelper

File: module/surveys/helper.js [hunk 7]
```diff
@@ -1888,4 +1886,44 @@ module.exports = class SurveysHelper {
...
+
+ /**
+      * Get imported surveys documents.
+      * @method
+      * @name getStartedSurveys
+      * @param  {String} userId - userId of user.
+      * @param  {String} programId - program Id.
+      * @returns {result} - all the survey which user has started in that program. 
+     */
+    static getStartedSurveys(userId, programId){
```
Comment (aks30) [inline module/surveys/helper.js:1898]: @ankitpws this will be moved to usersHelper

**Review by aks30 — CHANGES_REQUESTED** [review]
Reviewed on 31 May 11 AM

File: module/surveys/helper.js (hunk 7 above)
Comment (ankitpws) [reply module/surveys/helper.js]: Moved under usersHelper

File: module/observations/helper.js (hunk 6 above)
Comment (ankitpws) [reply module/observations/helper.js]: Moved under userHelper

File: controllers/v1/surveysController.js (hunk 5 above)
Comment (ankitpws) [reply controllers/v1/surveysController.js]: Created under users controller

File: controllers/v1/observationsController.js (hunk 4 above)
Comment (ankitpws) [reply controllers/v1/observationsController.js]: created under users controller

**Pull request merged**